usr/bin/irdata_text_transpose.py

```diff
@@ -75,47 +57,40 @@ def main():
     progname = os.path.basename(sys.argv[0])
 
     # Get the command line options.
-
-    syntax = cmdsyntax.Syntax(syntax_description)
-    try:
-        matches = syntax.get_args(sys.argv[1:])
-        args = matches[0]
-    except IndexError:
-        print("Syntax:", file=sys.stderr)
-        print(syntax_description, file=sys.stderr)
-        sys.exit(1)
-    else:
-        if "help" in args:
-            print(__doc__, file=sys.stderr)
-            print("Syntax:", file=sys.stderr)
-            print(syntax_description, file=sys.stderr)
-            sys.exit(1)
-
-    start_field = index_for_int(args.get("start-field", 1))
-    end_field = int_or_none(
-        args.get("end-field", None)
-    )  # treat 1-index as 0-index plus one
-
-    delimiter = args.get("delimiter", "\t")
-    delimiter_within_fields = args.get("delimiter-within-fields", "\t")
-
-    sequence_start = int_or_none(args.get("sequence-start"))
-
-    if "filename" in args:
-        filename_or_stream = args["filename"]
+    argparser = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    argparser.add_argument("-f", metavar="start-field", type=int, default=1)
+    argparser.add_argument("-t", metavar="end-field", type=int)
+    argparser.add_argument("-d", metavar="delimiter", default="\t")
+    argparser.add_argument("-w", metavar="delimiter-within-fields", default="\t")
+    argparser.add_argument("-s", metavar="sequence-start", type=int)
+    argparser.add_argument("filename", help="name of a file or '-'")
+
+    args = argparser.parse_args()
+
+    # treat 1-index as 0-index plus one
+    start_field = data.index_for_int(args.f)  # start-field
+    end_field = data.int_or_none(args.t)  # end-field
+    delimiter = args.d  # delimiter
+    delimiter_within_fields = args.w  # delimiter-within-fields
+    sequence_start = data.int_or_none(args.s)  # sequence-start
```

Comment:
Also probably breaks formatting rules.

```suggestion
    argparser.add_argument("-f", metavar="start-field", type=int, default=1, dest="start_field")
    argparser.add_argument("-t", metavar="end-field", type=int, dest="end_field")
    argparser.add_argument("-d", metavar="delimiter", default="\t", dest="delimiter")
    argparser.add_argument("-w", metavar="delimiter-within-fields", default="\t", dest="delimiter_within_fields")
    argparser.add_argument("-s", metavar="sequence-start", type=int, dest="sequence_start")
    argparser.add_argument("filename", help="name of a file or '-'")

    args = argparser.parse_args()

    # treat 1-index as 0-index plus one
    start_field = data.index_for_int(args.start_field)
    end_field = data.int_or_none(args.end_field)
    delimiter = args.delimiter
    delimiter_within_fields = args.delimiter_within_fields
    sequence_start = data.int_or_none(args.sequence_start)
```